### PR TITLE
MINOR: Tighten up locking when aborting expired transactions

### PR DESCRIPTION
**Pull request by apurvam**
This is a followup to #4137

### CODE REVIEW

**Comment by apurvam** [comment]
cc @hachikuji

File: core/src/test/scala/unit/kafka/utils/TestUtils.scala [hunk 1]
```diff
@@ -1382,13 +1382,21 @@ object TestUtils extends Logging {
...
+    */
+  def consumeRecordsFor[K, V](consumer: KafkaConsumer[K, V], duration: Long): Seq[ConsumerRecord[K, V]] = {
     val startTime = System.currentTimeMillis()
     val records = new ArrayBuffer[ConsumerRecord[K, V]]()
     waitUntilTrue(() => {
       records ++= consumer.poll(50).asScala
-      System.currentTimeMillis() - startTime > timeout
-    }, s"The timeout $timeout was greater than the maximum wait time.")
+      System.currentTimeMillis() - startTime > duration
+    }, s"The timeout $duration was greater than the maximum wait time.")
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/utils/TestUtils.scala:1399]: We should probably set the wait time to be longer than `duration` and then this error is not possible.

**Review by ijuma — COMMENTED** [review]
Thanks for the quick follow-up. Just a minor comment. Will leave it to @hachikuji to review the locking.

**Comment by ijuma** [comment]
@hachikuji does this look OK to you?

**Comment by hachikuji** [comment]
LGTM

File: core/src/test/scala/unit/kafka/utils/TestUtils.scala (hunk 1 above)
Comment (apurvam) [reply core/src/test/scala/unit/kafka/utils/TestUtils.scala:1399]: Actually the wait time is already the Long.MaxValue, so we can't be greater. But the api required an error message, so there it is.

**Pull request closed without merging**